edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/ModuleIdentityLifecycleManager.cs

```diff
@@ -104,12 +104,28 @@ public class ModuleIdentityLifecycleManager : IModuleIdentityLifecycleManager
             List<Module> updatedModulesIndentity = (await this.UpdateServiceModulesIdentityAsync(removeIdentities, createIdentities, updateIdentities)).ToList();
             ImmutableDictionary<string, Module> updatedDict = updatedModulesIndentity.ToImmutableDictionary(p => p.Id);
 
-            IEnumerable<IModuleIdentity> moduleIdentities = updatedModulesIndentity.Concat(modules.Where(p => !updatedDict.ContainsKey(p.Id))).Select(
-                p =>
-                {
-                    string connectionString = this.GetModuleConnectionString(p);
-                    return new ModuleIdentity(this.iothubHostName, this.gatewayHostName, this.deviceId, p.Id, new ConnectionStringCredentials(connectionString));
-                });
+            IEnumerable<IModuleIdentity> moduleIdentities;
+
+            if (!includeUnchangedIdentities)
+            {
+                moduleIdentities = updatedModulesIndentity.Concat(modules.Where(p => !updatedDict.ContainsKey(p.Id))).Select(
+                    p =>
+                    {
+                        string connectionString = this.GetModuleConnectionString(p);
+                        return new ModuleIdentity(this.iothubHostName, this.gatewayHostName, this.deviceId, p.Id, new ConnectionStringCredentials(connectionString));
+                    });
+            }
+            else
+            {
+                var allModules = await this.serviceClient.GetModules();
+                moduleIdentities = allModules.Select(
+                    p =>
+                    {
+                        string connectionString = this.GetModuleConnectionString(p);
+                        return new ModuleIdentity(this.iothubHostName, this.gatewayHostName, this.deviceId, p.Id, new ConnectionStringCredentials(connectionString));
```

Comment:
This seems out of date, like we should be getting the identities in the same way as `Edgelet/ModuleIdentityLifecycleManager.cs`. Can you double-check this?

Reply:
nevermind, I was looking at the wrong file.